# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/brush_mix_blend.glsl
# Commit: 0edb6b6f17b2
# Full Hash: 0edb6b6f17b24c737d053df19730e9c9e61e17ca
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r ee0de0eac18d -r 0edb6b6f17b2 gfx/wr/webrender/res/brush_mix_blend.glsl
--- a/gfx/wr/webrender/res/brush_mix_blend.glsl	Thu Dec 17 22:26:08 2020 +0000
+++ b/gfx/wr/webrender/res/brush_mix_blend.glsl	Thu Dec 17 22:26:10 2020 +0000
@@ -3,16 +3,15 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #define VECS_PER_SPECIFIC_BRUSH 3
+#define WR_FEATURE_TEXTURE_2D
 
 #include shared,prim_shared,brush
 
 // xy: uv coorinates.
-// z: layer.
-varying vec3 v_src_uv;
+varying vec2 v_src_uv;
 
 // xy: uv coorinates.
-// z: layer.
-varying vec3 v_backdrop_uv;
+varying vec2 v_backdrop_uv;
 
 flat varying int v_op;
 
@@ -41,16 +40,14 @@
     vec2 src_uv = src_device_pos +
                   src_task.common_data.task_rect.p0 -
                   src_task.content_origin;
-    v_src_uv.xy = src_uv / src_texture_size;
-    v_src_uv.z = src_task.common_data.texture_layer_index;
+    v_src_uv = src_uv / src_texture_size;
 
     RenderTaskCommonData backdrop_task = fetch_render_task_common_data(prim_user_data.y);
     float src_to_backdrop_scale = pic_task.device_pixel_scale / src_task.device_pixel_scale;
     vec2 backdrop_uv = device_pos +
                        backdrop_task.task_rect.p0 -
                        src_task.content_origin * src_to_backdrop_scale;
-    v_backdrop_uv.xy = backdrop_uv / backdrop_texture_size;
-    v_backdrop_uv.z = backdrop_task.texture_layer_index;
+    v_backdrop_uv = backdrop_uv / backdrop_texture_size;
 }
 #endif
 
@@ -213,8 +210,8 @@
 const int MixBlendMode_Luminosity  = 15;
 
 Fragment brush_fs() {
-    vec4 Cb = textureLod(sColor0, vec3(v_backdrop_uv.xy, v_backdrop_uv.z), 0.0);
-    vec4 Cs = textureLod(sColor1, vec3(v_src_uv.xy, v_src_uv.z), 0.0);
+    vec4 Cb = texture(sColor0, v_backdrop_uv);
+    vec4 Cs = texture(sColor1, v_src_uv);
 
     // The mix-blend-mode functions assume no premultiplied alpha
     if (Cb.a != 0.0) {